ada/ada

```diff
@@ -2215,14 +2214,22 @@ validate_input() {
             exit 1
           fi
           ;;
-        events | report-staged )
-          if [[ -z $channel_name || $channel_name =~ ^-- ]] ; then
-            echo 1>&2 "ERROR: command $command requires a channel name."
-            exit 1
-          fi
-          ;;
       esac
       ;;
+    events | report-staged )
+      if [[ -z $channel_name || $channel_name =~ ^-- ]] ; then
+        echo 1>&2 "ERROR: command $command requires a channel name and a directory."
+        exit 1
+      fi
+      if [[ -z $path || $path =~ ^-- ]] ; then
+        echo 1>&2 "ERROR: command $command requires a directory."
+        exit 1
+      fi
+      if ! is_dir "$path" ; then
```

Comment:
There is a problem with using `is_dir` here, before the `construct_auth` is called. You cannot execute API commands before the authorization is constructed for the curl command. Instead of throwing an error, `is_dir` returns 1 and the integration tests with `ada --events` fail because it thinks the directory does not exist.

Reply:
The next commit should fix that